From 5218ce360d510756ea1d9a7886138e09a3026427 Mon Sep 17 00:00:00 2001
Message-Id: <5218ce360d510756ea1d9a7886138e09a3026427.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: arsharma <ankitprasad.r.sharma@intel.com>
Date: Thu, 7 Aug 2014 16:25:21 +0530
Subject: [PATCH 173/312] MUST_REBASE [VPG]: drm/i915: Fix for Adjusting
 brightness causing flicker

In intel_panel_set_backlight, calling dpst_set_brightness for MIPI panel
if DPST is enabled, to ensure same backlight levels are being applied
by User and DPST.
This patch is marked MUST_REBASE because DPST will not be upstreamed

Issue: GMIN-2971
Change-Id: I998f53810abab2ccaa4a2f43ec325a0119e0c270
Signed-off-by: arsharma <ankitprasad.r.sharma@intel.com>
---
 drivers/gpu/drm/i915/intel_panel.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index f973afc..2bb15ce 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -636,7 +636,9 @@ void intel_panel_set_backlight(struct intel_connector *connector, u32 level,
 
 	mutex_unlock(&dev_priv->backlight_lock);
 
-	if (dev_priv->vbt.has_mipi)
+	if (dev_priv->vbt.has_mipi && dev_priv->dpst.enabled)
+		i915_dpst_set_brightness(dev, level);
+	else if (dev_priv->vbt.has_mipi)
 		intel_panel_actually_set_backlight(connector, level);
 }
 
-- 
1.7.9.5

